content/300-guides/150-testing/150-integration-testing.mdx

```diff
@@ -0,0 +1,450 @@
+
+# The containers that compose the project
+services:
+  db:
+    image: postgres:13
+    restart: always
+    container_name: integration-tests-prisma
+    ports:
+      - '5433:5432'
+    environment:
+      POSTGRES_USER: prisma
+      POSTGRES_PASSWORD: prisma
+      POSTGRES_DB: tests
+```
+
+> **Note**: The version used here is the latest at the time of writing, if you are following along be sure to use the same version for consistency.
+
+Breaking down the file:
+
+- The Postgres image is specified with the (version) 13 tag. This will be downloaded if you do not have it locally available.
+- The port `5433` is mapped to the internal (Postgres default) port `5432`. This will be the port number the database is exposed on externally.
+- The database user credentials are set and the database given a name.
+
+To connect to the database in the container you will need to create a new connection string. Given the credentials specified in the compose file that would look like the following:
+
+```env file=.env
+DATABASE_URL="postgresql://prisma:prisma@localhost:5433/tests"
+```
+
+To create the container in a detached state so that you can continue to use the terminal tab, run the following command from a terminal:
+
+```terminal
+docker compose up -d
+```
+
+Next you can check that the database has been created by executing a `psql` command inside the container.
+
+🆔 Get the containers id:
+
+<CodeWithResult expanded={true}>
+
+<Cmd>
+
+```
+docker ps
+```
+
+</Cmd>
+
+<CmdResult>
+
+```
+CONTAINER ID   IMAGE             COMMAND                  CREATED         STATUS        PORTS                    NAMES
+1322e42d833f   postgres:latest   "docker-entrypoint.s…"   2 seconds ago   Up 1 second   0.0.0.0:5433->5432/tcp   integration-tests-prisma
+```
+
+</CmdResult>
+
+</CodeWithResult>
+
+💾 Run `psql` in the container, login with the created user and check the database is created:
+
+<CodeWithResult expanded={true}>
+
+<Cmd>
+
+```
+docker exec -it 1322e42d833f psql -U prisma tests
+```
+
+</Cmd>
+
+<CmdResult>
+
+```
+tests=# \l
+                              List of databases
+   Name    | Owner  | Encoding |  Collate   |   Ctype    | Access privileges
+
+ postgres  | prisma | UTF8     | en_US.utf8 | en_US.utf8 |
+ template0 | prisma | UTF8     | en_US.utf8 | en_US.utf8 | =c/prisma        +
+           |        |          |            |            | prisma=CTc/prisma
+ template1 | prisma | UTF8     | en_US.utf8 | en_US.utf8 | =c/prisma        +
+           |        |          |            |            | prisma=CTc/prisma
+ tests     | prisma | UTF8     | en_US.utf8 | en_US.utf8 |
+(4 rows)
+```
+
+</CmdResult>
+
+</CodeWithResult>
+
+## Add a seed script
+
+Your integration tests will be run against a database in a test environment instead of your actual production or development environments.
+
+For the tests to best simulate real world scenarios you can seed the database on startup, thus providing your tests with some data to test against.
```

Comment:
I would emphasize realistic data more and cut 'thus' - e.g. "You can use a seed script to pre-populate the database with real world test data, such as names with special characters, and easily run your tests against the same dataset multiple times."

Reply:
Argh, i cant believe i used 'thus'. Shoot me next time please